src/components/TradeWidget/OrderDetails.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import styled from 'styled-components'
+import { useFormContext } from 'react-hook-form'
+
+import { TokenDetails } from 'types'
+import { FEE_PERCENTAGE } from 'const'
+
+const Wrapper = styled.dl`
+  margin: 2em 0 0 0;
+  font-size: 0.8em;
+`
+
+const Dd = styled.dd`
+  font-weight: bold;
+  margin: 0;
+`
+
+const Dt = styled.dt`
+  margin-left: 4em;
+`
+
+const Highlight = styled.span`
+  font-weight: bold;
+  color: #367be0;
+`
+
+// TODO: move to utils?
+function calculatePrice(sellAmount: number, receiveAmount: number): number {
+  return sellAmount > 0 ? receiveAmount / sellAmount : 0
+}
+
+interface Props {
+  sellTokenId: string
```

Comment:
I think this component should be more independent (more `dumb`:P ), so instead of depending on `sellTokenId` and watching the changes, I think it should have a `sellAmount`, `receiveAmount` props that are injected by its parent.

Reply:
Done